src/modules/adhs/src/lib/Adhs.c

```diff
@@ -0,0 +1,360 @@
+            }
+            status = EINVAL;
+        }
+
+        // Reset status and payload if TOML file could not be parsed or property not found, as it may yet have to be configured.
+        if (MMI_OK != status)
+        {
+            value = g_permissionConfigMapValues[0];
+            status = MMI_OK;
+        }
+    }
+
+    if (MMI_OK == status)
+    {
+        *payloadSizeBytes = strlen(value);
+
+        if ((g_maxPayloadSizeBytes > 0) && ((unsigned)*payloadSizeBytes > g_maxPayloadSizeBytes))
+        {
+            OsConfigLogError(AdhsGetLog(), "MmiGet(%s, %s) insufficient maxmimum size (%d bytes) versus data size (%d bytes), reported value will be truncated", componentName, objectName, g_maxPayloadSizeBytes, *payloadSizeBytes);
+            *payloadSizeBytes = g_maxPayloadSizeBytes;
+        }
+
+        *payload = (MMI_JSON_STRING)malloc(*payloadSizeBytes + 1);
+        if (NULL != *payload)
+        {
+            strncpy(*payload, value, *payloadSizeBytes);
+        }
+        else
+        {
+            OsConfigLogError(AdhsGetLog(), "MmiGet: failed to allocate %d bytes", *payloadSizeBytes + 1);
+            *payloadSizeBytes = 0;
+            status = ENOMEM;
+        }
+    }
+
+    if (IsFullLoggingEnabled())
+    {
+        OsConfigLogInfo(AdhsGetLog(), "MmiGet(%p, %s, %s, %.*s, %d) returning %d", clientSession, componentName, objectName, *payloadSizeBytes, *payload, *payloadSizeBytes, status);
+    }
+
+    return status;
+}
+
+int AdhsMmiSet(MMI_HANDLE clientSession, const char* componentName, const char* objectName, const MMI_JSON_STRING payload, const int payloadSizeBytes)
+{
+    int status = MMI_OK;
+
+    if ((NULL == componentName) || (NULL == objectName) || (NULL == payload) || (payloadSizeBytes < 0))
+    {
+        OsConfigLogError(AdhsGetLog(), "MmiSet(%s, %s, %p, %d) called with invalid arguments", componentName, objectName, payload, payloadSizeBytes);
+        status = EINVAL;
+        return status;
+    }
+
+    if (!IsValidSession(clientSession))
+    {
+        OsConfigLogError(AdhsGetLog(), "MmiSet(%s, %s) called outside of a valid session", componentName, objectName);
+        status = EINVAL;
+    }
+        
+    if ((MMI_OK == status) && (strcmp(componentName, g_adhsComponentName)))
+    {
+        OsConfigLogError(AdhsGetLog(), "MmiSet called for an unsupported component name (%s)", componentName);
+        status = EINVAL;
+    }
+        
+    if ((MMI_OK == status) && (strcmp(objectName, g_desiredOptInObjectName)))
+    {
+        OsConfigLogError(AdhsGetLog(), "MmiSet called for an unsupported object name (%s)", objectName);
+        status = EINVAL;
+    }
+        
+    if ((MMI_OK == status) && (!IsValidPayload(payload, payloadSizeBytes)))
+    {
+        OsConfigLogError(AdhsGetLog(), "MmiSet(%.*s, %d) called with invalid payload", payloadSizeBytes, payload, payloadSizeBytes);
+        status = EINVAL;
+    }
+
+    if (MMI_OK == status)
+    {
+        const char* value = NULL; 

```

Comment:
>const char* value = NULL;  [](http://example.com/codeflow?start=8&length=26)

C, do not define here. Here and elsewhere.

Reply:
Fixed, thanks!